xarray/core/parallelcompat.py

```diff
@@ -0,0 +1,199 @@
+        Called inside xarray.apply_ufunc, so must be supplied for vast majority of xarray computations to be supported.
+        """
+        ...
+
+    @staticmethod
+    def map_blocks():
+        """Called by xarray.map_blocks."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def blockwise():
+        """Called by some niche functions in xarray."""
+        raise NotImplementedError()
+
+
+class DaskManager(ChunkManager):
+    def __init__(self):
+        from dask.array import Array
+
+        self.array_type = Array
+
+    @staticmethod
+    def chunks(arr: "dask.array.Array"):
+        return arr.chunks
+
+    @staticmethod
+    def chunk(data: Any, chunks, **kwargs):
+        import dask.array as da
+
+        # dask-specific kwargs
+        name = kwargs.pop("name", None)
+        lock = kwargs.pop("lock", False)
+        inline_array = kwargs.pop("inline_array", False)
+
+        if is_duck_dask_array(data):
+            data = data.rechunk(chunks)
+        elif isinstance(data, DuckArrayModule("cubed").type):
+            raise TypeError("Trying to rechunk a cubed array using dask")
+        else:
+            if isinstance(data, indexing.ExplicitlyIndexed):
+                # Unambiguously handle array storage backends (like NetCDF4 and h5py)
+                # that can't handle general array indexing. For example, in netCDF4 you
+                # can do "outer" indexing along two dimensions independent, which works
+                # differently from how NumPy handles it.
+                # da.from_array works by using lazy indexing with a tuple of slices.
+                # Using OuterIndexer is a pragmatic choice: dask does not yet handle
+                # different indexing types in an explicit way:
+                # https://github.com/dask/dask/issues/2883
+                data = indexing.ImplicitToExplicitIndexingAdapter(
+                    data, indexing.OuterIndexer
+                )
+
+                # All of our lazily loaded backend array classes should use NumPy
+                # array operations.
+                dask_kwargs = {"meta": np.ndarray}
+            else:
+                dask_kwargs = {}
+
+            if utils.is_dict_like(chunks):
+                chunks = tuple(chunks.get(n, s) for n, s in enumerate(data.shape))
+
+            data = da.from_array(
+                data,
+                chunks,
+                name=name,
+                lock=lock,
+                inline_array=inline_array,
+                **dask_kwargs,
+            )
+        return data
+
+    @staticmethod
+    def rechunk(chunks, **kwargs):
+        ...
+
+    @staticmethod
+    def compute(arr, **kwargs):
+        return arr.compute(**kwargs)
+
+    @staticmethod
+    def apply_ufunc():
+        from dask.array.gufunc import apply_gufunc
+
+        ...
+
+    @staticmethod
+    def map_blocks():
+        from dask.array import map_blocks
+
+        ...
+
+    @staticmethod
+    def blockwise():
```

Comment:
Interestingly this is only used in `interp`. Potentially that could be refactored out to `dask_array_ops.py`

Reply:
Yeah I didn't even realise we called `blockwise` anywhere until making this PR. But in general we may want to call blockwise in other places in future, for example if I ever get around to finishing [merging xhistogram into xarray](https://github.com/pydata/xarray/pull/5400). :sweat_smile: